dids/src/main/kotlin/DIDIonManager.kt

```diff
@@ -0,0 +1,283 @@
+import web5.dids.ion.model.SidetreeCreateOperation
+import web5.dids.ion.model.toJWK
+import web5.dids.ion.model.toJsonWebKey
+import java.security.Provider
+import java.util.UUID
+
+private const val operationsPath = "/operations"
+private const val identifiersPath = "/identifiers"
+
+/**
+ * Configuration for the DIDIonManager.
+ *
+ * @property ionHost The ION host URL.
+ * @property updatePublicJsonWebKey The update public JSON Web Key. When absent, a new one will be generated.
+ * @property verificationPublicKey The verification public key. When absent, a new one will be generated.
+ * @property recoveryJsonWebKey The recovery JSON Web Key. When absent, a new one will be generated.
+ * @property engine The engine to use. When absent, a new one will be created from the [CIO] factory.
+ */
+public class DIDIonConfiguration internal constructor(
+  public var ionHost: String = "https://ion.tbddev.org",
+  public var updatePublicJsonWebKey: JsonWebKey? = null,
+  public var verificationPublicKey: PublicKey? = null,
+  public var recoveryJsonWebKey: JsonWebKey? = null,
+  public var engine: HttpClientEngine? = null,
+)
+
+
+/**
+ * Returns a DIDIonManager with the provided configuration [block].
+ */
+public fun DIDIonManager(block: DIDIonConfiguration.() -> Unit): DIDIonManager {
+  val conf = DIDIonConfiguration().apply(block)
+  return DIDIonManagerImpl(conf)
+}
+
+/** DIDIonCreator is sealed, so we provide an impl so the constructor can be called from the function above. */
+private class DIDIonManagerImpl(configuration: DIDIonConfiguration) : DIDIonManager(configuration)
+
+/**
+ * Base class for managing DIDIon operations. Uses the given [configuration].
+ */
+public sealed class DIDIonManager(
+  public val configuration: DIDIonConfiguration
+) {
+
+  @OptIn(ExperimentalSerializationApi::class)
+  private val json = Json {
+    prettyPrint = true
+    explicitNulls = false
+  }
+
+  private val operationsEndpoint = configuration.ionHost + operationsPath
+  private val identifiersEndpoint = configuration.ionHost + identifiersPath
+
+  private val engine: HttpClientEngine = if (configuration.engine == null) {
+    CIO.create {}
+  } else {
+    configuration.engine!!
+  }
+
+  private val client = HttpClient(engine) {
+    install(ContentNegotiation) {
+      json(json)
+    }
+  }
+
+  /**
+   * Creates a DID and DID Document.
+   *
+   * @return Pair of DID and DIDDocument.
+   */
+  public suspend fun create(): Triple<DID, DIDDocument, CreationMetadata> {
+    val createOp = createOperation()
+
+    val shortFormDIDSegment =
+      Base64URL.encode(
+        Multihash.sum(Multicodec.SHA2_256, canonicalized(createOp.suffixData)).get()?.bytes()
+      ).toString()
+    val initialState = InitialState(
+      suffixData = createOp.suffixData,
+      delta = createOp.delta,
+    )
+    val longFormDIDSegment = didUriSegment(initialState)
+
+    val response: HttpResponse = client.post(operationsEndpoint) {
```

Comment:
is this anchoring on every did we make? Maybe even in testing?

Should we have a flag pass through to use longform only instead of anchoring?

Reply:
weird